### PR TITLE
[BPK-3573] Add predefine calendar date styles

### PR DESCRIPTION
**Pull request by k0nserv**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

**Comment by georgegillams** [comment]
Approach seems good to me 👍

**Comment by imchristian** [comment]
LGTM 👌

File: Backpack/Calendar/Classes/BPKCalendar.h [hunk 1]
```diff
@@ -65,18 +102,32 @@ NS_SWIFT_NAME(CalendarDelegate) @protocol BPKCalendarDelegate<NSObject>
...
+ */
+- (BPKCalendarDateCellStyle)calendar:(BPKCalendar *)calendar cellStyleForDate:(BPKSimpleDate *)date;
+
 /**
  * Asks the delegate for a fill color for the specific date.
  *
  * @param calendar The backpack calendar.
  * @param date The date to provide fill-color for.
- * @return The fill/background colour for the given date.
+ * @return The fill/background color for the given date.
```
Comment (georgegillams) [inline Backpack/Calendar/Classes/BPKCalendar.h]:
```suggestion
 * @return The fill/background colour for the given date.
```

File: Backpack/Calendar/Classes/BPKCalendar.h [hunk 2]
```diff
@@ -85,8 +136,10 @@ NS_SWIFT_NAME(CalendarDelegate) @protocol BPKCalendarDelegate<NSObject>
  *
  * @param calendar The backpack calendar.
  * @param date The date to provide title-color for.
- * @return The title colour for the given date.
-*/
+ * @return The title color for the given date.
```
Comment (georgegillams) [inline Backpack/Calendar/Classes/BPKCalendar.h]:
```suggestion
 * @return The title colour for the given date.
```

File: Backpack/Calendar/Classes/BPKCalendar.m [hunk 3]
```diff
@@ -550,6 +578,48 @@ - (void)configureCell:(FSCalendarCell *)cell
...
+        case BPKCalendarDateCellStyleNormal:
+            return appearance.borderDefaultColor;
+        case BPKCalendarDateCellStylePositive:
+            return BPKColor.glencoe;
+        case BPKCalendarDateCellStyleNeutral:
+            return BPKColor.erfoud;
+        case BPKCalendarDateCellStyleNegative:
+            return BPKColor.hillier;
+        case BPKCalendarDateCellStyleCustom:
+            NSAssert(NO, @"Don't call `fillColorForDateStyle:appearance:` with the custom cell style");
```
Comment (georgegillams) [inline Backpack/Calendar/Classes/BPKCalendar.m]: This will only be called if we do something wrong in Backpack, right?

Comment (k0nserv) [reply Backpack/Calendar/Classes/BPKCalendar.m]: Yes it's to prevent and alert us if we do make such a mistake

Comment (georgegillams) [reply Backpack/Calendar/Classes/BPKCalendar.m]: I wonder if it should be more descriptive

Something like
`fillColorForDateStyle:appearance:` should not be called with the custom cell style. For a custom cell style, `calendar:fillColourForDate:` and `calendar:textColourForDate:` should be used instead.

Comment (k0nserv) [reply Backpack/Calendar/Classes/BPKCalendar.m]: I suppose it could

Comment (k0nserv) [reply Backpack/Calendar/Classes/BPKCalendar.m]: The important thing is that it crashes to alert the developer that they are about to create a bug

Comment (k0nserv) [reply Backpack/Calendar/Classes/BPKCalendar.m]: See 157fb04

**Comment by k0nserv** [comment]
image


Docs

**Pull request merged**